### PR TITLE
MGMT-2788 - fix installing hosts on day2 cloud cluster, when there is…

### PR DESCRIPTION
**Pull request by yevgeny-shnaidman**
… already an installed host present

### CODE REVIEW

File: internal/bminventory/inventory.go [hunk 1]
```diff
@@ -878,10 +878,20 @@ func (b *bareMetalInventory) refreshAllHosts(ctx context.Context, cluster *commo
 		return err
 	}
 	for _, chost := range cluster.Hosts {
-		if swag.StringValue(chost.Status) != models.HostStatusKnown && swag.StringValue(chost.Kind) != models.HostKindAddToExistingClusterHost {
+		if swag.StringValue(chost.Status) != models.HostStatusKnown && swag.StringValue(chost.Kind) == models.HostKindHost {
 			return common.NewApiError(http.StatusBadRequest, errors.Errorf("Host %s is in status %s and not ready for install",
 				hostutil.GetHostnameForMsg(chost), swag.StringValue(chost.Status)))
 		}
+
+		addHostsSkipStatuses := []string{
```
Comment (filanov) [inline internal/bminventory/inventory.go:886]: why do you need to filter the status here and not in the state machine?

File: internal/host/statemachine.go [hunk 2]
```diff
@@ -438,5 +438,17 @@ func NewHostStateMachine(th *transitionHandler) stateswitch.StateMachine {
 		})
 	}
 
+	// Noop transitions for day2 hosts
+	for _, state := range []stateswitch.State{
+		stateswitch.State(models.HostStatusAddedToExistingCluster),
```
Comment (filanov) [inline internal/host/statemachine.go:443]: Isn't the ticket talked about installed state? or did i miss something?

Comment (yevgeny-shnaidman) [reply internal/host/statemachine.go:443]: the installed status for day2 on cloud is AddedToExistingCluster

File: internal/host/statemachine.go [hunk 3]
```diff
@@ -438,5 +438,17 @@ func NewHostStateMachine(th *transitionHandler) stateswitch.StateMachine {
 		})
 	}
 
+	// Noop transitions for day2 hosts
+	for _, state := range []stateswitch.State{
```
Comment (filanov) [inline internal/host/statemachine.go:442]: You don't have to create a loop for only one transition.

Comment (yevgeny-shnaidman) [reply internal/host/statemachine.go]: fixed

**Review by filanov — APPROVED** [review]

**Pull request merged**